fsspec/spec.py

```diff
@@ -8,75 +10,176 @@
+
+    @_warn_implicit_alias
+    def move(self, path1, path2, **kwargs):
+        """Alias of :ref:`FilesystemSpec.mv`."""
+        if not self._add_aliases:
+            warnings.warn(self._alias_message.format("makedir"))
+        return self.mv(path1, path2, **kwargs)
+
+    @_warn_implicit_alias
+    def stat(self, path, **kwargs):
+        """Alias of :ref:`FilesystemSpec.info`."""
+        if not self._add_aliases:
+            warnings.warn(self._alias_message.format("makedir"))
+        return self.info(path, **kwargs)
+
+    @_warn_implicit_alias
+    def disk_usage(self, path, total=True, maxdepth=None, **kwargs):
+        """Alias of :ref:`FilesystemSpec.du`."""
+        if not self._add_aliases:
+            warnings.warn(self._alias_message.format("makedir"))
+        return self.du(path, total=total, maxdepth=maxdepth, **kwargs)
+
+    @_warn_implicit_alias
+    def rename(self, path1, path2, **kwargs):
+        """Alias of :ref:`FilesystemSpec.mv`."""
+        return self.mv(path1, path2, **kwargs)
+
+    @_warn_implicit_alias
+    def delete(self, path, recursive=False, maxdepth=None):
+        """Alias of :ref:`FilesystemSpec.rm`."""
+        return self.rm(path, recursive=recursive, maxdepth=maxdepth)
+
+    @_warn_implicit_alias
+    def upload(self, lpath, rpath, recursive=False, **kwargs):
+        """Alias of :ref:`FilesystemSpec.put`."""
+        return self.put(lpath, rpath, recursive=recursive, **kwargs)
+
+    @_warn_implicit_alias
+    def download(self, rpath, lpath, recursive=False, **kwargs):
+        """Alias of :ref:`FilesystemSpec.get`."""
+        return self.get(rpath, lpath, recursive=recursive, **kwargs)
 
 
 def make_instance(cls, args, kwargs):
     return cls(*args, **kwargs)
 
 
-class AbstractFileSystem(up):
+class _Cached(type):
+    """
+    Metaclass for caching file system instances.
+
+    Notes
+    -----
+    Instances are cached according to
+
+    * The values of the class attributes listed in `_extra_tokenize_attributes`
+    * The arguments passed to ``__init__``.
+
+    Filesystems are not cached beyond the usual lifetime of Python objects.
+    Deleting all references to a particular filesystem will result in it
+    being garbage collected.
+    """
+
+    cacheable = True
+    _extra_tokenize_attributes = ()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Note: we intentionally create a reference here, to avoid garbage
+        # collecting instances when all other references are gone. To really
+        # delete a FileSystem, the cache must be cleared.
+        self._cache = {}
+
+    def __call__(self, *args, **kwargs):
+        cls = type(self)
+        extra_tokens = tuple(
+            getattr(self, attr, None) for attr in self._extra_tokenize_attributes
+        )
+        token = tokenize(cls, *args, *extra_tokens, **kwargs)
+        if cls.cacheable and token in self._cache:
+            return self._cache[token]
```

Comment:
Is it a single global cache for all fsspec instances?

Reply:
No. The caching is per subclass type.